src/components/EmailLink.vue

```diff
@@ -0,0 +1,49 @@
+<script setup lang="ts">
+interface EmailLinkProps {
+	/**
+	 * Destination email address.
+	 */
+	to: string;
+	/**
+	 * Optional subject for the email.
+	 */
+	subject?: string;
+	/**
+	 * Optional body for the email.
+	 */
+	body?: string;
+}
+
+const {
+	subject = undefined,
+	body = undefined,
+	to,
+} = defineProps<EmailLinkProps>();
+
+let params = '';
+if (subject || body) {
+	const paramObj: {
+		subject?: string;
+		body?: string;
+	} = {};
+	if (subject) {
+		paramObj.subject = subject;
+	}
+	if (body) {
+		paramObj.body = body;
+	}
+	// The following line encodes both the key and value of the object using `encodeURIComponent`
+	// and joins it into a query param string. It's like `URLSearchParams` except spaces are
+	// encoded as `%20` instead of `+`, which is necessary for email clients to decode them.
+	// TODO: Extract this out to a utils file so we can document and test it.
```

Comment:
We don't have a lint rule to ban the use of `TODO` so I will have to emulate the rule!

If we wanna do this later then we should make an issue for it; otherwise just remove the note and let someone decide to do it later, OR even better: just do it now.

That said, I was asking the robots "hey does this really not exist?" and it flagged a bug in the implementation (this implementation is not able to support arrays apparently)

And then it was like `qs is a library that just does this` ✨  and so I'm gonna throw out the idea of adding https://www.npmjs.com/package/qs instead and using it for this.  With 95m weekly downloads we're in good company.